src/promptflow-evals/promptflow/evals/synthetic/README.md

```diff
@@ -1,66 +1,281 @@
 
-Install the `promptflow-evals` extra while installing promptflow.
+Install the `promptflow-evals` package.
+
+`pip install promptflow-evals`
 
 ## Usage
-Set the values for `AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_ENDPOINT` in the env or update this script:
 
-```
-import os
-from promptflow.evals.synthetic.qa import QADataGenerator, QAType, OutputStructure
+```python
+from promptflow.evals.synthetic import AdversarialSimulator
+from azure.identity import DefaultAzureCredential
+from typing import Any, Dict, List, Optional
+import asyncio
+
+
 
-os.environ["AZURE_OPENAI_API_KEY"] = ""
-os.environ['AZURE_OPENAI_ENDPOINT'] = ""
+azure_ai_project = {
+    "subscription_id": <subscription_id>,
+    "resource_group_name": <resource_group_name>,
+    "project_name": <project_name>,
+    "credential": DefaultAzureCredential(),
+}
 
-model_name = "gpt-4"
+async def callback(
+    messages: List[Dict],
+    stream: bool = False,
+    session_state: Any = None,
+    context: Dict[str, Any] = None
+) -> dict:
+    question = messages["messages"][0]["content"]
+    context = None
+    if 'file_content' in messages["template_parameters"]:
+        question += messages["template_parameters"]['file_content']
+    # the next few lines explains how to use the AsyncAzureOpenAI's chat.completions
+    # to respond to the simulator. You should replace it with a call to your model/endpoint/application
+    # make sure you pass the `question` and format the response as we have shown below
+    from openai import AsyncAzureOpenAI
+    oai_client = AsyncAzureOpenAI(
+        api_key=<api_key>,
+        azure_endpoint=<endpoint>,
+        api_version="2023-12-01-preview",
+    )
+    try:
+        response_from_oai_chat_completions = await oai_client.chat.completions.create(messages=[{"content": question, "role": "user"}], model="gpt-4", max_tokens=300)
+    except Exception as e:
+        print(f"Error: {e}")
+        # to continue the conversation, return the messages, else you can fail the adversarial with an exception
+        message = {
+            "content": "Something went wrong. Check the exception e for more details.",
+            "role": "assistant",
+            "context": None,
+        }
+        messages["messages"].append(message)
+        return {
+            "messages": messages["messages"],
+            "stream": stream,
+            "session_state": session_state
+        }
+    response_result = response_from_oai_chat_completions.choices[0].message.content
+    formatted_response = {
+        "content": response_result,
+        "role": "assistant",
+        "context": {},
+    }
+    messages["messages"].append(formatted_response)
+    return {
+        "messages": messages["messages"],
+        "stream": stream,
+        "session_state": session_state,
+        "context": context
+    }
 
-model_config = dict(
-    deployment=model_name,
-    model=model_name,
-    max_tokens=2000,
+```
+
+### Adversarial QA:
+```python
+scenario = "adv_qa"

```

Comment:
should this be enum from AdversarialSimulator class?

Reply:
yes, changed it.